src/app/infobox/infobox.component.spec.ts

```diff
@@ -73,7 +37,34 @@ describe('InfoboxComponent', () => {
     fixture.detectChanges();
   });
 
-  it('should create', () => {
+  it('should create Infobox component', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should have infobox heading same as query', () => {
+    let compiled = fixture.debugElement.nativeElement;
+
+    expect(compiled.querySelector('h2.heading')).toBeTruthy();
```

Comment:
these occur only when there are results. so this won't work.
workaround:-
assign some mock object to this.results and then do fixture.detectChanges()
then this will work